datumaro/cli/contexts/project/__init__.py

```diff
@@ -399,6 +402,52 @@ def filter_command(args):
 
     return 0
 
+def build_generate_parser(parser_ctor=argparse.ArgumentParser):
+    parser = parser_ctor(help="Generate synthetic dataset",
+        description="""
+        This command is useful for making a synthetic dataset
+        with specified shape and storing them in provided directory.|n
+        |n
+        Examples:|n
+        - Generate 300 3-channel synthetic images with H=224, W=256 and store to data_dir:|n
+        |s|s%(prog)s -o data_dir -k 300 --shape 224 256|n
+        """,
+        formatter_class=MultilineFormatter)
+
+    parser.add_argument('-o', '--output-dir', required=True,
+        help="Output directory to store generated dataset")
+    parser.add_argument('-k', '--count', type=int, required=True,
+        help="Number of images to be generated")
+    parser.add_argument('--shape', nargs='+', type=int,
+        help="Data shape separated by a space")
```

Comment:
It's not really correct to describe it as being space-separated, because the different dimensions are just separate arguments. A space-separated argument would look like `--shape='100 200'`.